fsmonitor.c

```diff
@@ -185,10 +185,12 @@ static int query_fsmonitor(int version, const char *last_update, struct strbuf *
 int fsmonitor_is_trivial_response(const struct strbuf *query_result)
 {
 	static char trivial_response[3] = { '\0', '/', '\0' };
-	int is_trivial = !memcmp(trivial_response,
-				 &query_result->buf[query_result->len - 3], 3);
 
-	return is_trivial;
+	if (query_result->len < 3)
+		return 0;
+
+	return !memcmp(trivial_response,
+		       &query_result->buf[query_result->len - 3], 3);
```

Comment:
Interesting. Under what circumstances did you experience this?

Also, it might look more in line with Git's coding conventions to do this instead:
```suggestion
	return query_result->len >= 3 &&
		!memcmp(trivial_response,
		        &query_result->buf[query_result->len - 3], 3);
```

Reply:
I've only seen this in tests, specifically here in t7519 (although there are multiple cases that trigger the same issue) - I have no idea if this happens in real life usage:
https://github.com/git/git/blob/966e671106b2fd38301e7c344c754fd118d0bb07/t/t7519-status-fsmonitor.sh#L235

Thanks for the style advice - I'll update my patch as suggested!